scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt

```diff
@@ -34,35 +37,71 @@ import java.util.TreeSet
 
 abstract class AbstractScannerTest : StringSpec() {
     protected val config = ScannerConfiguration()
-    private val licenseFilePath = File("../LICENSE")
+    private val commonlyDetectedFiles = listOf("LICENSE", "LICENCE", "COPYING")
+    private lateinit var inputDir: File
     private lateinit var outputDir: File
 
     abstract val scanner: LocalScanner
     abstract val expectedFileLicenses: TreeSet<String>
+    abstract val expectedDirectoryLicenses: TreeSet<String>
     abstract val testTags: Set<Tag>
 
     override fun beforeSpec(spec: Spec) {
         super.beforeSpec(spec)
+        inputDir = createTempDir(ORT_NAME, javaClass.simpleName)
+
+        // Copy our own root license under different names to a temporary directory so we have something to operate on.
+        val ortLicense = File("../LICENSE")
+
+        // This is loosely based on the patterns from
+        // https://github.com/licensee/licensee/blob/6c0f803/lib/licensee/project_files/license_file.rb#L6-L43.
+        commonlyDetectedFiles.forEach { ortLicense.copyTo(inputDir.resolve(it)) }
+    }
+
+    override fun beforeTest(testCase: TestCase) {
+        super.beforeTest(testCase)
         outputDir = createTempDir(ORT_NAME, javaClass.simpleName)
     }
 
-    override fun afterSpec(spec: Spec) {
+    override fun afterTest(testCase: TestCase, result: TestResult) {
         outputDir.safeDeleteRecursively()
+        super.afterTest(testCase, result)
+    }
+
+    override fun afterSpec(spec: Spec) {
+        inputDir.safeDeleteRecursively()
         super.afterSpec(spec)
     }
 
     init {
         "Scanning a single file succeeds".config(tags = testTags) {
             val resultsFile = outputDir.resolve("${scanner.scannerName}.${scanner.resultFileExt}")
 
-            val result = scanner.scanPath(licenseFilePath, resultsFile)
+            val result = scanner.scanPath(inputDir.resolve("LICENSE"), resultsFile)
 
             resultsFile.isFile shouldBe true
             result.summary.fileCount shouldBe 1
             result.summary.licenses shouldBe expectedFileLicenses
-            result.summary.licenseFindings.all {
-                it.location.path == licenseFilePath.invariantSeparatorsPath
-            } shouldBe true
+
+            // Check for relative paths in the result.
+            result.summary.licenseFindings.forEach {
```

Comment:
How about re-writing the assertion in a way which makes the code comment unnecessary? Like e.g.

`result.summary.licenseFindings.map { it.location.path} shouldContainExactlyInAnyOrder listOf(
   "relpath/LICENSE", 
     ...  
)`

Reply:
Without the comment, I find it to be less clear what property of the string the assertion is checking for. For the single file the string would be just `"LICENSE"` which does not really tell the reader that this should be a relative path.

But maybe I'll still play with the `listOf()` idea as it also asserts for the number of license findings, which is nice.

Reply:
After looking at the code again, I've simply removed the comment and blank line as `shouldNotStartWithPath` indeed makes it clear that paths should be relative.